docs/Guides-Renderer.md

```diff
@@ -0,0 +1,168 @@
+
+```
+ReactDOM.render(
+  <Relay.Renderer
+    Container={ProfilePicture}
+    queryConfig={profileRoute}
+  />,
+  component
+);
+```
+
+When the **Relay.Renderer** above is rendered, Relay will construct a query and send it to the GraphQL server. As soon as all required data has been fetched, `ProfilePicture` will be rendered. Props with fragments will contain data that was fetched from the server.
+
+If either `Container` or `queryConfig` ever changes, **Relay.Renderer** will immediately start attempting to fulfill the new data requirements.
+
+**Relay.Renderer** renders the loading state whenever it cannot immediately fulfill data needed to render. This often happens on the initial render, but it can also happen if either `Container` or `queryConfig` changes.
+
+By default, nothing is rendered while loading data for the initial render. If a previous set of `Container` and `queryConfig` were fulfilled and rendered, the default behavior is to continue rendering the previous view.
+
+## render prop
+
+We can change this behavior by supplying the `render` prop:
+
+```
+<Relay.Renderer
+  Container={ProfilePicture}
+  queryConfig={profileRoute}
+  render={({props}) => {
+      if (props) {
+        return <ProfilePicture {...props} />;
+      } else {
+        return <div>Loading...</div>;
+      }
+  }}
+/>
+```
+
+This snippet configures **Relay.Renderer** to render the "Loading..." text whenever it needs to fetch data.
+
+The `render` callback is called with an object with the following properties:
+
+```
+render({
+  props: ?{[propName: string]: mixed},
+  done: boolean,
+  error: ?Error,
+  retry: ?Function,
+  stale: boolean
+}): ?React$Element
+```
+
+### props
+
+```
+props: ?{[propName: string]: mixed}
+```
+
+If present, sufficient data is ready to render the container. This object must be spread into the container using the spread attribute operator. If absent, there is insufficient data to render the container.
+
+### done
+
+```
+done: boolean
+```
+
+Whether all data dependencies have been fulfilled. If `props` is present but `done` is false, then sufficient data is ready to render, but some data dependencies have not yet been fulfilled.
+
+### error
+
+```
+error: ?Error
+```
+
+If present, an error occurred while fulfilling data dependencies. If `props` and `error` are both present, then sufficient data is ready to render, but an error occurred while fulfilling deferred dependencies.
+
+### retry
+
+```
+retry: ?Function
+```
+
+A function that can be called to re-attempt to fulfill data dependencies. This property is only present if an `error` has occurred.
+
+### stale
+
+```
+stale: boolean
+```
+
+True when `forceFetch` is enabled and `props` is present due to available client data. Once the forced server request completes, `stale` will return to false.
+
+If a `render` callback is not supplied, the default behavior is to render the container if data is available, the existing view if one exists, or nothing.
```

Comment:
Since this isn't part of the `stale` docs, we'll need to move this text around. Maybe it belongs above with the preamble under `## Render callback`?